test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FullUriFunctionalTests.cs

```diff
@@ -71,52 +71,63 @@ public void SelectOrExpandCanOnlyBeCalledOnEntity()
             parseWithNonEntity.ShouldThrow<ODataException>().WithMessage(ODataErrorStrings.UriParser_TypeInvalidForSelectExpand("Edm.String"));
         }
 
-        [Fact(Skip = "This test currently fails.")]
+        [Fact]
         public void ParseSelectExpand()
         {
             ODataUriParser parser = new ODataUriParser(HardCodedTestModel.TestModel, new Uri("http://www.odata.com/OData"), new Uri("http://www.odata.com/OData/Dogs?$select=Color, MyPeople&$expand=MyPeople"));
             ODataUri parsedUri = parser.ParseUri();
-            parsedUri.SelectAndExpand.SelectedItems.First().ShouldBePathSelectionItem(new ODataPath(new PropertySegment(HardCodedTestModel.GetDogColorProp())));
-            var myPeopleExpand = parsedUri.SelectAndExpand.SelectedItems.Last().ShouldBeSelectedItemOfType<ExpandedNavigationSelectItem>().And;
+            parsedUri.SelectAndExpand.SelectedItems.Count().Should().Be(3);
+            parsedUri.SelectAndExpand.SelectedItems.ElementAt(1).ShouldBePathSelectionItem(new ODataPath(new PropertySegment(HardCodedTestModel.GetDogColorProp())));
+            var myPeopleExpand = parsedUri.SelectAndExpand.SelectedItems.First().ShouldBeSelectedItemOfType<ExpandedNavigationSelectItem>().And;

```

Comment:
q: don't understand why verification on Last() is changed to First(). Can we add verification related to Last() as  well? 
IE Browser cannot view the whole line of the code, limiting max length to 120-ish might be helpful?